### ISSUE TITLE
Two factor login flow

### ISSUE THREAD

**Issue by dansup** [tootsuite/mastodon#9782]
Logged into wrong account with 2FA enabled, never entered 2FA code and tried to login to a different account without success.

### Expected behaviour
- Login to account A
- At 2FA screen, realize I meant to login to account B
- Refresh page without entering 2FA code for account A
- Login to account B
- Successfully logged in

### Actual behaviour
- Login to account A
- At 2FA screen, realize I meant to login to account B
- Refresh page without entering 2FA code for account A
- Try to login to account B
- Error says "invalid credentials" even though account B's credentials are valid.
- Logged into account A and entered 2FA
- Logged out of account A
- Logged into account B, successfully

### Steps to reproduce the problem
See above, this happened on mastodon.social

**Comment by nightpool** [comment]
I believe that to be the case from the expected behavior, yes

**Comment by dansup** [comment]
@HankG Both accounts have 2FA enabled

**Comment by nightpool** [comment]
you were able to log into account B without entering a two factor code? are
you sure?

**Comment by dansup** [comment]
@nightpool No that is not what I meant. The problem is that after entering the correct credentials for an account, you have to proceed with 2FA or else you can't log into any other account.

**Comment by nightpool** [comment]
right, got that, I think me and the previous commenter were both just
confused by the way you phrased your account B login steps
